### ISSUE TITLE
Type Error in normal training

### ISSUE THREAD

**Issue by YangHan-Morningstar** [tensorflow/addons#2455]
Hi, I have use the tfa.layers.CRF in my BiLSTM_CRF, it could be compiled but I got an Error which says: "TypeError Expected int32 got 0.0 of type 'float' instead "

My target is [batch_size, seq_len] which type is "int32" and I use the first outputs of the CRF layer, which [batch_size, seq_len]. Now I'm confused about it, could you help me? Thanks a lot!

Here is my code :

![image](https://user-images.githubusercontent.com/67748964/115678380-046a0680-a384-11eb-9d13-c11b320208c0.png)
![image](https://user-images.githubusercontent.com/67748964/115678419-0f249b80-a384-11eb-9e0d-65ffe7fd3776.png)

**Comment by luozhouyang** [comment]
The SparseCategoricalCrossnentropy expect logits as prediction outputs, whose dtype should be a float, but `crf_decode_outputs` is a sequence of int. You should use `potentials` to compute the loss .

Here is an example of how to build a CRFModel: https://github.com/luozhouyang/transformers-keras/blob/0751e6c3bf48ceb8773a5bf957149fcc93f3537b/transformers_keras/token_classification/crf_models.py#L19

And here is a tests: https://github.com/luozhouyang/transformers-keras/blob/0751e6c3bf48ceb8773a5bf957149fcc93f3537b/tests/token_classification_tests/crf_models_test.py#L16

**Comment by howl-anderson** [comment]
Hi @YangHan-Morningstar I just submit a  **WIP** tutorial for CRF layer at #2552 . It will show you how to use CRF layer. If you still have any questions about how to use CRF layer after read the tutorial, please let me know it.

**Comment by howl-anderson** [comment]
Hi @YangHan-Morningstar if my tutorial in the previous comment works for you, please close this PR for me. If it does not work for you or if you have any other question about your code, please submit in the comment so we have the chance to solve it. Thank you!

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
